functions/index.js

```diff
@@ -276,7 +276,120 @@ exports.deleteUsersInAuthAndDB = functions.https.onCall(async (data, context) =>
     }
     // 暫定対応
     // データ数が100件ほどになると、処理が落ちることがあるため、sleepさせる
-    sleep(1000)
+    await sleep(1000)
   })
   return res
 })
+
+// 本番用データの登録
+exports.registerProdData = functions.https.onCall(async (data, context) => {
+  const db = admin.firestore()
+  const getAuth = admin.auth()
+  const res = { email: data.email, name: data.name }
+
+  await db
+    .collection('users')
+    .where('status', '==', 'test')
```

Comment:
ごめんまたまた、追加での要望で申し訳ない・・
ここのクエリはテスト用のデータの学生とその学生の年度で絞れるようにして欲しいかな！

仮に、年度が違うけど、学籍番号・Eメール・氏名が同じ学生がいたら、もともとの年度でパスワードが更新されることになるし！（留年の学生だとこのケースが出てきそう）

```suggestion
    .where('status', '==', 'test')
    .where('year', '==', data.year)
```

複数のクエリを使うときは、複合インデックスていうのをFirebaseに登録しないとエラーになるから注意して！

Reply:
ここちょっと試してみたんやけど、
テストの手動入力で入力するデータを2098年度にして、その後に、本番の手動入力で入力するデータ名前とかは同じにして、年度を2099年度にしたら、何もデータ追加されへんからちょっと、その原因を調べて欲しいです！

正しい挙動としては、新しい年度で同じデータの学生が追加されるはず・・

Reply:
これは同じid(メールアドレス)で複数のAuthenticationユーザを作成できないからだと思います

なのでそもそも、「年度だけ違う同じユーザ」は登録できなさそうですね…
どうしましょう？

Reply:
あー、そうか。。
そんな仕様やったなあ、確かに。。

そうなると、同じメードアドレスの学生がいる場合は、DBとAuthからユーザを削除する処理を追加して欲しいかな！
209行目からの処理を参考にして欲しい！